src/climsight/climsight_engine.py

```diff
@@ -370,6 +381,62 @@ def llm_request(content_message, input_params, config, api_key, stream_handler):
     #yield f"Generating..."
     logging.info(f"Generating...")    
 
+    ## === RAG integration === ##
+    # langchain embeddig model
+    langchain_ef = OpenAIEmbeddings(openai_api_key=api_key, model=EMBEDDING_MODEL) # max_retries, request_timeout, retry_min_seconds
+    # load & retrieve database 
+    db = Chroma(persist_directory=CHROMA_PATH, embedding_function=langchain_ef, collection_name="ipcc-collection")
+    print(f"There are {db._collection.count()} documents in the collection.")
+
+    retriever = db.as_retriever()
+
+    def format_docs(docs):
+        return "\n\n".join(doc.page_content for doc in docs)
+
+    template = """You are an assistant for retrieving information from IPCC reports for 
+    a particular location that is given to you. Use the following pieces of context 
+    to answer the question at the end.
+    Focus on retrieving the most relevant and concise information that 
+    addresses the question. If you cannot find information for that particular region,
+    you are allowed to extend the answer to something more general. However, in that case
+    you have to make clear that it is a general answer and does not apply particularly to 
+    the given region / coutry. Avoid including unrelated details. 
+    Consider the geographical location mentioned.
+    If you don't know the answer, just say that you don't know, don't try to make up an answer.
+
+    Content from IPCC reports: {context}
+    Location: {location}
+    Question: {question}
+
+    Helpful Answer:"""
+
+    custom_rag_prompt = PromptTemplate.from_template(template)
+
+    # work-around to be able and pass the location as sth executable to the chain 
+    # (see https://github.com/langchain-ai/langchain/discussions/16421)
+    location = input_params['location_str']
+    def get_loci(_):
+        return location
+    
+    # inspect chain - just for development
+    def inspect(state):
+        """Print the state passed between Runnables in a langchain and pass it on"""
+        print(state)
+        return state
```

Comment:
Do we need it at all ?

Reply:
Makes debugging easier. Left it, just changed from print to logger.info.